### ISSUE TITLE
If a comment is placed in the middle of the definition, SCSS will be corrupted when Stylelint automatically corrects it.

### ISSUE THREAD

**Issue by xootani** [stylelint/stylelint#6944]
### What steps are needed to reproduce the bug?

``` scss
$test: (
  "light" 0.25, // test
  "dark": #ffffff, // test
) !default;

$test2: (
// test
  "light-test": $test,
// test2
  "dark-test" : $test
) !default;
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-twbs-bootstrap"
  ],
  "rules": {
    "scss/dollar-variable-default": [
      true,
      {
        "ignore": "local"
      }
    ],
    "scss/selector-no-union-class-name": true,
    "scss/dollar-variable-pattern": "^[a-z]+([a-z0-9-_]+[a-z0-9]+)?$",
    "selector-max-class": null,
    "length-zero-no-unit": null
  }
}
```

### How did you run Stylelint?

``` shell
stylelint ".\scss\Untitled-1.scss" --fix
```

### Which version of Stylelint or dependencies are you using?

stylelint 15.8.0
stylelint-config-twbs-bootstrap 10.0.0

### What did you expect to happen?

``` scss
$test: (
  "light" .25, // test
  "dark": #fff, // test
) !default;

$test2: (
  // test
  "light-test": $test,
  // test2
  "dark-test" : $test
) !default;
```

### What actually happened?

Stylelint output is here.

``` scss
$test: (
  "light" .25, 
  "dark": #ffffff, 
) !default;

$test2: (
  
  "light-test": $test,

  "dark-  test" : $test
) !default;
```

Console  
```
scss/Untitled-1.scss
 2:15  ✖  Unexpected whitespace at end of line  stylistic/no-eol-whitespace
 3:19  ✖  Unexpected whitespace at end of line  stylistic/no-eol-whitespace
 7:2   ✖  Unexpected whitespace at end of line  stylistic/no-eol-whitespace

3 problems (3 errors, 0 warnings)
```

**Comment by ybiquitous** [comment]
@xootani Thanks for the report using the template. Unfortunately, I cannot reproduce it with [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkAXGBnF8AEAKAOgHbbb4gA2AlgOYAWKZ2AdAEwCsANNgPTfZqYiJMlACGAJwDWZHAGIAZoq69+GBoQCU2AISx5ogK7kUAbiJFUaljgLEefAeuEUa9ALSOZ2S5g5D7qpgs-iISkh5qjDg+6lq6MPpGpkQgHCDylOQwAHKiALZwiDAAHvkADllM6GDo6KngEIQZ1AggwCEgJWiEUOheANr+zpgAnllUhChukE00HgDuAEbobosQECiY4qJlZP4Aun52ZOJGGF7tdsM16NxQEOTkEm4AbhKUootZbnqGxgNDEj8U4wI5AoGXcHgsg0QgQcQwLxkcgQMCich7K5AgC+gMOgLI1Vq3HQMCyYBQ8LccLcBkIlEa0yetWp+URCGBBlBBJARNu90ezze4g+XxgbjKohQaHEhCRIAAev1RG4AF77ADUuGVaoADG4AJxuAD6mp1qv1Bs1GgA-EgyGCgYSyTAKVS8qJikzRLUvIQjORHc4soRqChaGqYOIINSY3TKAwOf7Hv5cYRsfUZs0AGLwj0oVoAK3QjXqsDKdUQkNIvJQYzJlEm8oAjGwmAAOJi6h0dUbjRtTLNzFBLFZrDZbHYt3VdruYjPYoA). Do I miss something?

**Comment by xootani** [comment]
@ybiquitous 
Thanks for taking alook at my report.

I think the reason why the problem is not reproduced in the demo is because the contents of example.scss are incorrect.
If you have Example.scss with the following contents, the problem can be reproduced.
``` scss
$test: (
  "light" 0.25, // test
  "dark": #ffffff, // test
) !default;

$test2: (
// test
  "light-test": $test,
// test2
  "dark-test" : $test
) !default;
```

**Comment by ybiquitous** [comment]
Thanks. Now I can reproduce the problem with [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkAXGBnF8AEAKAOgHbbb4gA2AlgOYAWKZ2ADAHQBMArADTYD0v2NJiIkyUAIYAnANZkcAYgBmylT36CMDQgEpsAQliLxAV3IoA3ESKpNbHAULqhW0RRr0AtM7nYbmLkROtiKkIBIyXpqMOH5augYwRqYWRCBcIIqU5DAAcuIAtnCIMAAeBQAO2SzoYOjoaeAQhJnUCCDAIWSlaIRQ6D4A2iGumACe2VSEKB6QzTReAO4ARugeSxAQKJiS4uVkIQC6AcShkqYYPh0nI7XovFAQ5ORSHgBuUpTiS9kehiZmg2GJEEZxgx2BwKuEIhZBohAgkhgPjI5AgYHE5H212BAF8gUcgWQanVeOgYNkwCgER54R5jIRKE0Zs86jSCkiECDjGDCSBiXcHk8Xu9JJ9vjAPOVxCg0JJCMiQAA9AbiDwALwOAGpcCr1UwPABODwAfS1urV+oNWu0AH4kGRwcCieSYJTqflxCVmeI6j5CKZyI7XNlCNQULR1TBJBAaTH6ZQGJz-U8QnjCDiGrMWgAxBEelBtABW6CaDVg5XqiChoTGE0oUwVAEYOCwABwsJgOzp8lDjcn16ZZ+YoZardabba7JusGdYjM4oA)!

We need to identify which rule's autofix unexpectedly removes `//` comments. 🤔

**Comment by ybiquitous** [comment]
@xootani Could you please help identify the cause rule? We want to know it from the built-in rules or 3rd-party rules.

**Comment by jeddy3** [comment]
It [appears to be some interplay](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkAXGBnF8AEAKAOgHbbb4gA2AlgOYAWKZ2ADAHQBMArADTYD0AVNjSZs-XkRJkoAQwBOAazI4AxADN1Gnr15CMDQgEpsAQlirpAV3IoA3ESKo9bHAULbdmCaQo16AWmEGBGxHTC4id0C2Lyk5eQC9RhxQlFJDEzNLaztCe0JpbGAvAAdZCGKXLwFsSABbWphCVLFsADdpcj5edS8tQTqGptFecOI2jq8DIgBfEC4QVUpyGAA5aQaEEBgAD3Xi5ZZ0MHR0OfAIQkXqTaLiMjALTAhagGUATybpbaVvYohMY7oPxHE5kUa-cgWaiUQjoH4AbTImDeyyoTWBKBRlEwlDAZAAuuCyLIrBgfrcSJIQMiqDiwLwYbBPihKBcftFiNMZmdIJcaAAxCCyWrSFCbABW6AuZ1gxVOiAp3hpMDRQRwZAAjBwWAAOFhMMExamY1EwlAYrF0n5kfiGsZkP4Ak7AwHWkC2kDc6ZAA) between the `stylistic/indentation` custom rule and the `postcss-scss` custom syntax. 

As both are 3rd-party, we can close the issue in our tracker.

@xootani I suspect you'll need to keep digging. Perhaps open an issue tracker in the [`stylelint-stylistic`](https://github.com/elirasza/stylelint-stylistic) plugin, which the `stylelint-config-twbs-bootstrap` config uses under the hood. The issue maybe resolved there or with the `postcss-scss` custom syntax. It's difficult to determine which without a deeper dive.
